docs/docs-content/clusters/public-cloud/azure/azure-disk-encryption.md

```diff
@@ -0,0 +1,353 @@
+---
+sidebar_label: "Azure Disk Encryption"
+title: "Azure Disk Encryption"
+description:
+  "How to enable disk encryption on your Azure IaaS cluster using Disk Encryption Sets with customer-managed keys"
+icon: ""
+hide_table_of_contents: false
+tags: ["public cloud", "azure", "encryption", "security"]
+sidebar_position: 12
+---
+
+Palette supports disk encryption of your Azure Kubernetes cluster using Disk Encryption Sets with
+[customer-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#customer-managed-keys).
+
+By default, Azure encrypts all managed disks with
+[platform-managed keys](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#platform-managed-keys),
+however, customer-managed keys enable you to have greater control over your key management.
+
+You can use Disk Encryption Sets to encrypt your nodes' Operating System (OS) and data disks by including a preset
+option in the <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" />
+pack.
+
+## Limitations
+
+- Azure Disk Encryption is available from [4.5.a](../../../release-notes/release-notes.md) onwards.
+
+- Azure Disk Encryption is only supported on Azure IaaS clusters.
+
+- Azure Disk Encryption is only supported when using the
+
+  <VersionedLink text="Palette eXtended Kubernetes (PXK)" url="/integrations/packs/?pack=kubernetes" /> pack.
+
+- If a key expires in your Key Vault, your cluster may experience operation failures. To resolve this, generate a new
+  key in Key Vault and update your Disk Encryption Set to reference the new key.
+
+  - No changes are needed in Palette, as the Palette eXtended Kubernetes pack configuration references the URI of your
+    Disk Encryption Set, which remains unchanged.
+
+  - We recommend enabling
+    [**Auto key rotation**](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption#automatic-key-rotation-of-customer-managed-keys)
+    on your Disk Encryption Set so it can automatically use new key versions from your Key Vault.
+
+- Changing the Disk Encryption Set URI (`diskEncryptionSetID`) in the Palette eXtended Kubernetes pack configuration in
+  Palette will trigger a node repave.
+
+## Prerequisites
+
+- To create the Azure Key Vault and Disk Encryption Set, you will need an Azure user account with the **Contributor** or
+  **Owner** role on the resource group or subscription.
```

Comment:
We do not recommend the contributor role. We have gone to great lengths to document all the required Azure [permissions](https://deploy-preview-4558--docs-spectrocloud.netlify.app/clusters/public-cloud/azure/required-permissions/) so using the contributor is frowned upon. So if there are permission they need then let's call them out, but let's not tell people to use the overprivileged Contributor role because they will use it a point back to this doc.

Reply:
They mention this in the Feature Overview, but I will break this down into permissions required for each task.